src/libcollections/string.rs

```diff
@@ -670,6 +670,122 @@ impl String {
     pub fn clear(&mut self) {
         self.vec.clear()
     }
+
+    /// Creates a draining iterator that clears the specified range in the String
+    /// and iterates over the characters contained in the range.
+    ///
+    /// # Example
```

Comment:
Also I thought the convention for Strings was that indices were always byte-wise?